Only document modules for public API

aaronrenner opened the pull request:
This removes quite a few modules from the API docs that aren't meant for public usage. This will allow us to know exactly what functions people depend on and which ones we need to consider backward compatibility on. Here's what the API docs look like now.

![screen shot 2017-04-26 at 1 59 30 pm](https://cloud.githubusercontent.com/assets/120878/25454945/4bc5f212-2a8b-11e7-9e6d-657bed4eec79.png)

There's still quite a few of deprecated functions in `Wallaby.Browser` that need to be removed, but I figured they will be taken care of as part of #177.

aaronrenner commented:
I also ran across [this link](https://hexdocs.pm/elixir/1.4.2/writing-documentation.html#hiding-internal-modules-and-functions) in the Elixir docs about hiding modules that aren't part of the public api.

PragTob commented on lib/wallaby.ex:
```diff
@@ -36,7 +37,35 @@ defmodule Wallaby do
+
+    assert user1 |> find(@message_list) |> List.last |> text == "Hello yourself"
+    assert user2 |> find(@message_list) |> List.first |> text == "Hello there"
+  end
```
Code examples :heart_eyes:

PragTob commented on lib/wallaby/dsl.ex:
````diff
@@ -1,5 +1,26 @@
+  ## Example
+
+  ```
+  defmodule MyPageObject do
````
I know that coming from Ruby one tends to call them page objects, but they aren't really as it's just a module. I would just call it `MyPage` - what do you think?

keathley replied:
I second that idea.

aaronrenner replied:
That sounds good. I'll check in a commit shortly.

PragTob commented on lib/wallaby/dsl.ex:
````diff
@@ -1,5 +1,26 @@
+      |> click(Query.button("Save"))
+    end
+  end
+  ```
````
Thanks for adding these comprehensive examples! :+1:

PragTob commented on lib/wallaby/helpers/key_codes.ex:
```diff
@@ -1,8 +1,5 @@
-  Helper utility for converting key atoms into key codes sutiable to send over
-  the wire.
-  """
+  @moduledoc false
```
While we already have the comment, maybe we keep it as a code comment? I'm not a big fan of those, but a high level comment for the module might help for new contributors? Wdyt?

PragTob commented on lib/wallaby/metadata.ex:
```diff
@@ -1,9 +1,5 @@
-  Metadata is used to encode information about the browser and test. This
-  information is then stored in a User Agent string. The information from the
-  test can then be extracted in the application.
-  """
```
While we already have the comment, maybe we keep it as a code comment? I'm not a big fan of those, but a high level comment for the module might help for new contributors? Wdyt?

keathley replied:
Whats the standard here? I'm OK with leaving these sorts of docs simply because they help other contributors.

PragTob replied:
According to the link aaron posted to the elixir docs using code comments for private functions is the standard:

> ##  Documenting Private Functions
>
> Elixir warns if a private function has a @doc attribute and discards its content, because @doc is intended to be used only for your public interface.
>
> Private functions may still need internal documentation for maintainers, though. That can be accomplished with code comme

aaronrenner replied:
I felt bad deleting that documentation, so I think a code comment is a great idea. 👍

PragTob commented on lib/wallaby/phantom.ex:
```diff
@@ -1,13 +1,45 @@
+
+  ## Notes
+
+  This driver requires PhantomJS be installed in your path. You can install PhantomJS through NPM or your package manager of choice:
```
unless you specify the phantomjs path (which is what I usually do, all to the same npm package so everyone runs the same version) - maybe we could make that clearer here?

keathley replied:
We could update the example below this to show how to use the phantom in your node modules

aaronrenner replied:
@PragTob I'm not sure I understand what these docs should look like. I copied these notes from the readme and have an example of setting the `:phantomjs` config value on line 24. What would be a good way to improve the clarity here?

PragTob replied:
I'd say that it requires it in the path or you configure a custom path. As for the example I think it's something like `node_modules/bin/phantomjs` or something :)

PragTob commented on lib/wallaby/phantom.ex:
```diff
@@ -41,63 +75,96 @@ defmodule Wallaby.Phantom do
   defdelegate find_elements(session_or_element, compiled_query),  to: Driver
+  @doc false
   defdelegate send_keys(session_or_element, keys),                to: Driver
+  @doc false
```
a `doc @false start/end` would be useful here :D

keathley commented on lib/wallaby/dsl.ex:
```diff
@@ -1,5 +1,26 @@
+      |> visit("/registration.html")
+      |> fill_in(Query.text_field("Name"), with: "Chris")
+      |> fill_in(Query.text_field("email"), with: "c@keathly.io")
+      |> click(Query.button("Save"))
```
One other addition that would be good to add here is to move these queries into module attributes or functions. Thats a pattern that I think we should be actively trying to encourage. What do y'all think?

PragTob replied:
Totally :+1: from my side

aaronrenner replied:
I just updated this to use module attributes and match the rest of the documentation. I'm not sure why, but dialyzer throws a fit when I store these queries in module attributes in my personal project. It could be just me, though.

```
def set_email(session, email) do
  session
  |> fill_in(Query.css("[name=email]"), with: email)
end
```
works, but 
```
@email_field Query.css("[name=email]")

def set_email(session, email) do
  session
  |> fill_in(@email_field, with: email)
end
```
gives me the following warning
```
Function set_email/2 has no local return
```

keathley commented on lib/wallaby/session.ex:
```diff
@@ -1,44 +1,5 @@
 defmodule Wallaby.Session do
-  @moduledoc """
```
Glad this is finally being removed 👍

aaronrenner commented:
I just checked in the suggested updates, but wasn't sure what to do for [doc changes](https://github.com/keathley/wallaby/pull/204#discussion_r113864849) in `lib/wallaby/phantom.ex`.

PragTob commented:
Just commented on those :)

aaronrenner commented:
Just updated the example phantomjs path to point to `node_modules/.bin/phantomjs`. I think I've incorporated all of the feedback so far. Let me know if you'd like any more changes.

PragTob commented:
Well let's get this party in then! Thanks! :)
